### PR TITLE
Update the budget page to be printable

### PR DESCRIPTION
**Pull request by jfdoming**
I've seen this request a few times and got curious what it'd take. Originally I tried the Reports page which seems more difficult, but the Budget page seems doable. Feedback welcome! I've only tested this in Chrome—would appreciate help testing in other browsers.

Suggested testing steps:
1. Add a bunch of categories to the budget.
2. Hit Ctrl-P to print the page and see what it looks like on various configurations.

### CODE REVIEW

**Comment by Teprifer** [comment]
Using my real budget. yay privacy mode working in the print.

Firefox - defaulted to 1 page by condensing it down. Seems overall it thinks the page is much wider than it is?
image
image


100% scale is, probably best not done heh.
image

Snippet of pages 2 and 3: 
image

Chome has it neatly centered and fine etc. and splits over two pages, Edge is similar, both also show a yellow underline for the demo build notice of the netlify link that Firefox doesn't:
image

Firefox will render background colours correctly:


Chrome doesn't render background colours per the theme used, note the dark theme underneath the print window(right side), but light in the print window:
image

This works correctly in firefox:
image

**Comment by youngcw** [comment]
Any idea on why the print is insisting on starting in October instead of using the current view?  It will add more months if I add them, as long as I only want to look at 3 at a time.  Any more than 3 and I will still just get 3 in the print preview

File: packages/desktop-client/src/hooks/useSpaceMeasure.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useRef } from 'react';
+
+import { View } from '../components/common/View';
+
+export const useSpaceMeasure = () => {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/hooks/useSpaceMeasure.tsx:5]: Would existing `useViewportSize`/`useWindowSize` hooks achieve the same thing as this hook?

**Comment by joel-jeremy** [comment]
Just some thought on the implementation. Would a `PrivacyFilter` style component work to specify which components are visible on print or not? e.g.

```js
<PrintAware overflow="visible">
    <MyComponent />
</PrintAware>
```

PrintAware component would just wrap it's  children with a div that has `@media print` className. This would allow us to declaratively configure how components should behave during printing.

**Review by MikesGlitch — COMMENTED** [review]
This looks cool. 

I noticed the budget rows have a larger width than the header - replicated on Edge and firefox:

image

Also When I select the 5 month display and try to print on Edge it only prints 3 months in the portait. On Firefox it prints 5. It works fine if you print to landscape.

I think these are minor though.

**Pull request closed without merging**